### DISCUSSION TITLE
.umirc.ts配置metas-viewport不生效

### DISCUSSION THREAD

**Discussion by jiangjjiang** [umijs/umi#8535] [Q&A]
`  metas: [
    {
      name: 'viewport',
      content: 'width=device-width,initial-scale=1,maximum-scale=1,minimum-scale=1,user-scalable=no',
    },
  ],`

**Comment by fz6m** [comment]
Create `${projectRoot}/plugin.ts` :

```ts
// plugin.ts

import { IApi } from 'umi';

export default (api: IApi) => {
  api.modifyHTML(($) => {
    $('meta[name=viewport]').attr('content', `width=device-width,initial-scale=1,maximum-scale=1,minimum-scale=1,user-scalable=no`)
    return $;
  });
};
```

**Reply by jiangjjiang** [reply]
还是无法生效，在添加该配置文件之前就看到页面控制台中viewport被注释了，但项目代码中没有这一部分内容。
![image](https://user-images.githubusercontent.com/109278940/179187517-028d0a56-5304-425d-8f50-466c022678ba.png)

**Reply by fz6m** [reply]
umi3 ？旧版本的问题自己寻找方法解决下吧，或者 PR 。

**Comment by fz6m** [comment]
将在下个版本中修复，目前可以使用如上插件修改 html 的方式解决。

FYI：

 - https://github.com/umijs/umi/pull/8539
